### PR TITLE
Pp 845 i os add unit tests for skonto ra

### PR DESCRIPTION
**Pull request by mrkulik**

### CODE REVIEW

File: BankSDK/GiniBankSDK/Tests/GiniBankSDKTests/Skonto/SkontoViewModelTests.swift [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+        viewModel.setExpiryDate(newDate)
+        XCTAssertEqual(viewModel.remainingDays,
+                       remainingDays,
+                       "Remaining days should be recalculated correctly when the expiry date changes.")
+    }
+    
+    func testExpiredDiscountEdgeCase() {
+        let pastDate = Calendar.current.date(byAdding: .day, value: -1, to: Date()) ?? Date()
+        viewModel.setExpiryDate(pastDate)
+        XCTAssertEqual(viewModel.edgeCase,
```
Comment (ValentinaIancu-Gini) [inline BankSDK/GiniBankSDK/Tests/GiniBankSDKTests/Skonto/SkontoViewModelTests.swift:90]: [suggestion]: We can also add this last assertion: 
```
   XCTAssertFalse(viewModel.isSkontoApplied, "Skonto should not be applied when the discount is expired.")
```

File: BankSDK/GiniBankSDK/Tests/GiniBankSDKTests/Skonto/PriceTextFieldTests.swift [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+        super.setUp()
+        priceTextField = PriceTextField(frame: .zero)
+        priceTextField.priceDelegate = self
+        priceTextField.text = formatValue(0)
+    }
+
+    override func tearDown() {
+        priceTextField = nil
+        priceTextChangeExpectation = nil
+        super.tearDown()
```
Comment (ValentinaIancu-Gini) [inline BankSDK/GiniBankSDK/Tests/GiniBankSDKTests/Skonto/PriceTextFieldTests.swift:25]: [clarification] Should we add also `priceTextField.priceDelegate = nil` inside tearDown method?

**Review by ValentinaIancu-Gini — COMMENTED** [review]
@mrkulik Nice work.
I've added some small comments. Thank you!

File: BankSDK/GiniBankSDK/Tests/GiniBankSDKTests/Skonto/PriceTextFieldTests.swift [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+
+    func priceTextField(_ textField: PriceTextField, didChangePrice editedText: String) {
+        priceTextChangeExpectation?.fulfill()
+    }
+
+    func testSingleDigitInput() {
+        let input = "1"
+        let expectedValue = formatValue(0.01)
+        let expectation = expectation(description: "Price text change should be triggered")
+        priceTextChangeExpectation = expectation
```
[the remark this answers is not in the document]
Comment (mrkulik) [reply BankSDK/GiniBankSDK/Tests/GiniBankSDKTests/Skonto/PriceTextFieldTests.swift:36]: Its done only in first test to be sure that delegate will be triggered and simulate input method works correctly in tests

File: BankSDK/GiniBankSDK/Tests/GiniBankSDKTests/Skonto/PriceTextFieldTests.swift (hunk 2 above)
Comment (mrkulik) [reply BankSDK/GiniBankSDK/Tests/GiniBankSDKTests/Skonto/PriceTextFieldTests.swift:25]: We can skip it, because delegate is weak inside textField

**Review by ValentinaIancu-Gini — APPROVED** [review]
@mrkulik nice job!

**Pull request merged**